### ISSUE TITLE
torch backpropagate problem

### ISSUE THREAD

**Issue by cuge1995** [ajhamdi/AdvPC#4]
Hey, I'm trying to use torch to reimplement your work. But occurring the backpropagate problem.
`RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [256]] is at version 5; expected version 4 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).`

the code is as follows:
```python
logits = self.model(adv_data)
new_logits = self.model(self.encoder(adv_data)))
adv_loss = 0.75*self.adv_func(logits, target).mean() + 0.25*self.adv_func(new_logits, target).mean()
```
Do you think the code is right?

**Comment by ajhamdi** [comment]
yes, it should be like that. This error happens when the tensor in the Pytorch graph is being modified

**Comment by ajhamdi** [comment]
@cuge1995 could you please direct me to the repo of the PyTorch implementation. I can help with that.
